### ISSUE TITLE
Unable to cast object of type

### ISSUE THREAD

**Issue by aradzys** [HewlettPackard/POSH-HPOneView#32]
Hello,

I wonder if you can help me.

Connect-HPOVMgmt gives an error 

unknown error [Cannot create object of type "HPOneView.PKI.SslCertificate". Unable to cast object of type 'System.Management.Automation.PSObject' to type 
'System.String[]'.].

Powershell V5
Latest HPonview POSH release

Thanks,

Arturas

**Comment by ChrisLynchHP** [comment]
Can you do the following:

* Start a new PowerShell console, or clear the $Error (e.g. $Error.Clear()) variable before continuing.
* Attempt your Connect-HPOVMgmt call again
* If the error generates, please capture the output from the following:
``` PowerShell
$error | % { 
   write-host $_.message
   write-host $_.ScriptStackTrace
}
```

I can't seem to replicate your issue, and it might be a trapped error that isn't being handled correctly.

**Comment by aradzys** [comment]
Hi Chris,

thanks for quick reply. Here is the output of an error:

$error | % { 
   write-host $_.message
   write-host $_.ScriptStackTrace
}

at Connect-HPOVMgmt<Begin>, C:\windows\system32\windowspowershell\v1.0\Modules\HPOneView.120\HPOneView.120.psm1: line 2201
at <ScriptBlock>, <No file>: line 1

at Send-HPOVRequest<Begin>, C:\windows\system32\windowspowershell\v1.0\Modules\HPOneView.120\HPOneView.120.psm1: line 1434
at Connect-HPOVMgmt<Begin>, C:\windows\system32\windowspowershell\v1.0\Modules\HPOneView.120\HPOneView.120.psm1: line 2187
at <ScriptBlock>, <No file>: line 1

at Show-HPOVSSLCertificate<Process>, C:\windows\system32\windowspowershell\v1.0\Modules\HPOneView.120\HPOneView.120.psm1: line 5372
at Send-HPOVRequest<Begin>, C:\windows\system32\windowspowershell\v1.0\Modules\HPOneView.120\HPOneView.120.psm1: line 1410
at Connect-HPOVMgmt<Begin>, C:\windows\system32\windowspowershell\v1.0\Modules\HPOneView.120\HPOneView.120.psm1: line 2187
at <ScriptBlock>, <No file>: line 1

**Comment by ChrisLynchHP** [comment]
Hmm...  That didn't really provide me what I need.  Can you do the following and provide the output:

``` PowerShell
$error | % { 
   write-host $_
   write-host $_.ScriptStackTrace
}
```

**Comment by aradzys** [comment]
Here you go :) I've changed hostname as you can notice. I'm abe to ping
host and also get connected from test PC on the same network using
powershell version4

Unable to connect to 'xxxx1.xxx.net' due to unknown error [Cannot create
object of type "HPOneView.PKI.SslCertificate". Unable to cast object of
type 'System.Management.Automation.PSObject' to type 'System.String[]'.].
at Connect-HPOVMgmt<Begin>,
C:\windows\system32\windowspowershell\v1.0\Modules\HPOneView.120\HPOneView.120.psm1:
line 2201
at <ScriptBlock>, <No file>: line 1
Unable to connect to xxxx1.xxx.net' due to unknown error [Cannot create
object of type "HPOneView.PKI.SslCertificate". Unable to cast object of
type 'System.Management.Automation.PSObject' to type 'System.String[]'.].
at Send-HPOVRequest<Begin>,
C:\windows\system32\windowspowershell\v1.0\Modules\HPOneView.120\HPOneView.120.psm1:
line 1434
at Connect-HPOVMgmt<Begin>,
C:\windows\system32\windowspowershell\v1.0\Modules\HPOneView.120\HPOneView.120.psm1:
line 2187
at <ScriptBlock>, <No file>: line 1
Cannot create object of type "HPOneView.PKI.SslCertificate". Unable to cast
object of type 'System.Management.Automation.PSObject' to type
'System.String[]'.
at Show-HPOVSSLCertificate<Process>,
C:\windows\system32\windowspowershell\v1.0\Modules\HPOneView.120\HPOneView.120.psm1:
line 5372
at Send-HPOVRequest<Begin>,
C:\windows\system32\windowspowershell\v1.0\Modules\HPOneView.120\HPOneView.120.psm1:
line 1410
at Connect-HPOVMgmt<Begin>,
C:\windows\system32\windowspowershell\v1.0\Modules\HPOneView.120\HPOneView.120.psm1:
line 2187
at <ScriptBlock>, <No file>: line 1

**Comment by ChrisLynchHP** [comment]
Thanks.  Can you also provide the output from $PSVersionTable?

Also, are you using the default Self-Signed Certificate, or have you created a new CA signed cert on your appliance?

**Comment by aradzys** [comment]
Hi,

I'm using certificate from our domain CA. Strange that module works on
version 4 where same conditions apply.

Here is version table:

$PSVersionTable


Name  : PSVersion
Value : 5.0.10240.16384

Name  : WSManStackVersion
Value : 3.0

Name  : SerializationVersion
Value : 1.1.0.1

Name  : CLRVersion
Value : 4.0.30319.42000

Name  : BuildVersion
Value : 10.0.10240.16384

Name  : PSCompatibleVersions
Value : {1.0, 2.0, 3.0, 4.0...}

Name  : PSRemotingProtocolVersion
Value : 2.3

**Comment by ChrisLynchHP** [comment]
Does your Windows 10 system trust your domain CA root?

I’m not sure why the behavior would be different on Windows 10 versus your other PowerShell 4 system.  I’m curious, can you capture and send the following to chris dot lynch (at) hp (dot) com, from a PowerShell console where you have not loaded or used the HP OneView library?

``` PowerShell

$ConnectString = "YourAppliance”
$ConnectString = "https://$Appliance"
$WebRequest = [System.Net.HttpWebRequest]::Create($ConnectString)

#Attempt connection to appliance.
try { $Response = $WebRequest.GetResponse() }
catch [System.Net.WebException] { Write-host $_.Exception.Status }

    $Cert = [Security.Cryptography.X509Certificates.X509Certificate2]$WebRequest.ServicePoint.Certificate.Handle

    try { $SAN = ($Cert.Extensions | Where-Object {$_.Oid.Value -eq "2.5.29.17"}).Format(0) -split ", " }
    catch { $SAN = $null }

    $chain = New-Object Security.Cryptography.X509Certificates.X509Chain 

    [void]$chain.ChainPolicy.ApplicationPolicy.Add("1.3.6.1.5.5.7.3.1")
    $Status = $chain.Build($Cert)

    Write-Host $chain.ChainStatus
    Write-Host $Status
```

**Comment by aradzys** [comment]
Thing is I got same error with powershell version 5 using these OS :
Windows 7 SP1 64 bit, Windows 8.1 64bi and Winows 10 bit. All were
enterprise editions.

Not sure I follow this "Does your Windows 10 system trust your domain CA
root?" do you mean if root CA is installed on Win 10 PC?

I will send email shortly

**Comment by ChrisLynchHP** [comment]
I was asking if your PowerShell 5 systems trusted the root cert CA.  I just loaded an internal CA cert, and have still yet to reproduce your issue.

**Comment by ChrisLynchHP** [comment]
I have resolved this issue in the [1.20.240.0 release](https://github.com/HewlettPackard/POSH-HPOneView/releases/tag/v1.20.240.0).
